src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerRemoveCmdletBase.cs

```diff
@@ -0,0 +1,123 @@
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using ResourceModel = Microsoft.Azure.Management.EdgeGateway.Models.Trigger;
+using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeTrigger;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Triggers
+{
+    [Cmdlet(VerbsCommon.Remove, Constants.Trigger, DefaultParameterSetName = DeleteByNameParameterSet,
+         SupportsShouldProcess = true
+     ),
+     OutputType(typeof(bool))]
+    public class DataBoxEdgeTriggerRemoveCmdletBase : AzureDataBoxEdgeCmdletBase
+    {
+        private const string DeleteByNameParameterSet = "DeleteByNameParameterSet";
+        private const string DeleteByInputObjectParameterSet = "DeleteByInputObjectParameterSet";
+        private const string DeleteByResourceIdParameterSet = "DeleteByResourceIdParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DeleteByResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.ResourceIdHelpMessage,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = DeleteByInputObjectParameterSet,
+            HelpMessage = Constants.InputObjectHelpMessage,
+            Position = 0
+        )]
+        [ValidateNotNull]
+        public PSResourceModel InputObject { get; set; }
+
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = DeleteByNameParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = DeleteByNameParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
+        public string DeviceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = DeleteByNameParameterSet,
+            HelpMessage = Constants.NameHelpMessage,
+            Position = 2)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.PassThruHelpMessage)]
+        public SwitchParameter PassThru;
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.AsJobHelpMessage)]
+        public SwitchParameter AsJob { get; set; }
+
+        private bool Remove()
```

Comment:
`Remove` is not clearly enough for a function name. It will be better to named as `RemoveDataBoxEdgeTrigger`.

Reply:
This class name itself is DataBoxEdgeTriggerRemove and all the remove classes for different cmdlets can follow the same pattern